test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/condition/OpenShiftODOProjectExists.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.condition;
+
+import org.eclipse.reddeer.common.condition.AbstractWaitCondition;
+import org.eclipse.reddeer.common.exception.RedDeerException;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftApplicationExplorerView;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShiftODOConnection;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShiftODOProject;
+
+/**
+ * Wait condition to wait for OpenShift ODO Project exists.
+ * 
+ * @author jkopriva@redhat.com
+ *
+ */
+public class OpenShiftODOProjectExists extends AbstractWaitCondition {
+	
+	private String projectName;
+
+	/**
+	 * Constructs OODOProjectIsDeleted wait condition. Condition is met when project is deleted.
+	 * 
+	 * @param projectName project name
+	 */
+	public OpenShiftODOProjectExists(String projectName) {
+		this.projectName = projectName;
+	}
+	
+	@Override
+	public boolean test() {
+		try {
+			OpenShiftApplicationExplorerView explorer = new OpenShiftApplicationExplorerView();
+			explorer.open();
+			OpenShiftODOConnection connection = explorer.getOpenShiftODOConnection();
+			connection.refresh();
```

Comment:
I just know how it worked in OC openshift tooling, if you had some issues when refreshing the connection it might have thrown an error dialog. Maybe (but I am not sure how it works in odo) it might be wise to add some error dialog handling... But that's your call ;)

Reply:
@jkopriva No changes here?